lib/LevelMaker/levelMaker.cpp

```diff
@@ -0,0 +1,57 @@
+#include "levelMaker.h"
+
+//check if item collides with other items
+bool checkCollision(Item item, Item itemsArray[], int amountOfItems)
+{
+    for (int i = 0; i < amountOfItems; i++)
+    {
+        if (item.x < itemsArray[i].x + itemsArray[i].size &&
+            item.x + item.size > itemsArray[i].x &&
+            item.y < itemsArray[i].y + itemsArray[i].size &&
+            item.y + item.size > itemsArray[i].y)
+        {
+            return true;
+        }
+    }
+    return false;
+}
```

Comment:
should this not just be in game logic?

Reply:
I would argue it has nothing to do with the gamelogic, as the gamelogic does the calculating and not the initializing of the game. As we want to expand on the levelMaking for different stages of the game. So we can have level 1, 2, and 3 with different logic. So all of that logic could be in this class. And the gamelogic class is already huge. Isn't it better to kinda split of the code a bit?

Reply:
well you can always split the game logic but isnt this the function that checks collisions every game tick? or is this only used in the initialization of the game?

Reply:
it's only used when creating the items. It's a check to make sure a new item isn't placed on top of another item. That's why it's in the generateItems class. It's only called in the generateItem function